hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/customfees/AdjustmentUtils.java

```diff
@@ -198,8 +203,21 @@ public static void adjustHbarFees(final AssessmentResult result, final AccountID
         final var fixedSpec = hbarFee.fixedFee();
         if (fixedSpec != null) {
             final var amount = fixedSpec.amount();
-            hbarAdjustments.merge(sender, -amount, Long::sum);
-            hbarAdjustments.merge(collector, amount, Long::sum);
+            hbarAdjustments.merge(sender, -amount, AdjustmentUtils::addExactOrThrow);
+            hbarAdjustments.merge(collector, amount, AdjustmentUtils::addExactOrThrow);
+        }
+    }
+
+    public static long addExactOrThrow(final long a, final long b) {
+        return addExactOrThrowReason(a, b, INSUFFICIENT_SENDER_ACCOUNT_BALANCE_FOR_CUSTOM_FEE);
+    }
+
+    public static long addExactOrThrowReason(
+            final long a, final long b, @NonNull final ResponseCodeEnum failureReason) {
+        try {
+            return Math.addExact(a, b);
+        } catch (final ArithmeticException ignore) {
+            throw new HandleException(INSUFFICIENT_SENDER_ACCOUNT_BALANCE_FOR_CUSTOM_FEE);
```

Comment:
Should we be using `failureReason` here ?

Reply:
Ah good catch Neeha, fixed.